xquant/keras/keras_report_utils.py

```diff
@@ -0,0 +1,380 @@
+        """
+        Convert Keras model to networkx graph representation.
+
+        Args:
+            model: Keras model to convert.
+            repr_dataset: Representative dataset (not used in Keras).
+
+        Returns:
+            Graph representing the model.
+        """
+        graph = graph_preparation_runner(in_model=model,
+                                         representative_data_gen=repr_dataset,
+                                         quantization_config=DEFAULTCONFIG,
+                                         fw_info=DEFAULT_KERAS_INFO,
+                                         fw_impl=KerasImplementation(),
+                                         tpc=DEFAULT_KERAS_TPC)
+        return graph
+
+    def wrapped_dataset(self,
+                        dataset: Any,
+                        is_validation: bool,
+                        device: str = None) -> Any:
+        """
+        Generator function that wraps 'dataset' to be able to handle both
+        representative and validation datasets.
+
+        Args:
+            dataset (Any): The dataset to wrap.
+            is_validation (bool): Flag indicating if this is a validation dataset.
+            device (str, optional): Device to use for inference. Defaults to None.
+
+        Yields:
+            Any: Processed data from the dataset.
+        """
+
+        def process_data(x: Any, is_validation: bool) -> Any:
+            return x[0] if is_validation else x
+
+        for x in dataset():
+            yield process_data(x, is_validation)
+
+    def get_float_to_quantized_compare_points(self,
+                                              quantized_model: keras.Model,
+                                              float_model: keras.Model) -> Dict[str, str]:
+        """
+        Get comparison points between the floating-point and quantized models.
+
+        Args:
+            quantized_model (keras.Model): The quantized model.
+            float_model (keras.Model): The floating-point model.
+
+        Returns:
+            Dict[str, str]: A dictionary mapping comparison points between the two models.
+        """
+        quant_points_names = [
+            layer.name for layer in quantized_model.layers
+            if isinstance(layer, KerasQuantizationWrapper)
+        ]
+
+        float_name2quant_name = {}
+
+        for quant_point in quant_points_names:
+            candidate_float_layer_name = quantized_model.get_layer(quant_point).layer.name
+
+            if candidate_float_layer_name in [layer.name for layer in float_model.layers]:
+                if candidate_float_layer_name not in float_name2quant_name:
+                    float_name2quant_name[candidate_float_layer_name] = quant_point
+                else:
+                    Logger.critical(f"Duplicate mapping found for layer: {candidate_float_layer_name}")
+            else:
+                Logger.warning(f"Skipping point {quant_point}")
```

Comment:
"Could not find a matching layer in the float model to layer with name {}, skipping it in similarity metrics comparison points computation."